tests/integration/test_api.py

```diff
@@ -386,3 +386,68 @@ def test_annotation_event(client, monkeypatch, httpserver):
             },
         )
     assert result.status_code == 200
+
+
+def test_annotate_insight_anonymous_then_authenticated(client):
+    """Test that annotating first as anonymous, then, just after, as authenticated validate the anotation"""
+
+    # first the user validates annotation without being connected
+    result = client.simulate_post(
+        "/api/v1/insights/annotate",
+        params={
+            "insight_id": insight_id,
+            "annotation": 1,
+            "device_id": "voter1",
+        },
+    )
+
+    assert result.status_code == 200
+    assert result.json == {
+        "description": "the annotation vote was saved",
+        "status": "vote_saved",
+    }
+
+    # For non-authenticated users we expect the insight to not be validated, with only a vote being cast.
+    votes = list(AnnotationVote.select())
+    assert len(votes) == 1
+
+    insight = next(
+        ProductInsight.select()
+        .where(ProductInsight.id == insight_id)
+        .dicts()
+        .iterator()
+    )
+
+    assert not any(insight[key] for key in ("username", "completed_at", "annotation"))
```

Comment:
Maybe I missed it but you should have added "process_after" here to the tuple

```suggestion
    assert not any(insight[key] for key in ("username", "completed_at", "annotation", "process_after"))
```